PayloadPkg/FirmwareUpdate/FirmwareUpdateHelper.c

```diff
@@ -590,6 +591,72 @@ IsUpdateComponentForContainer (
    return TRUE;
 }
 
+/**
+  Perform Config data svn check
+
+  This function will perform svn checks for cfgdata in flash and
+  config data in capsule.
+
+  @param[in]  ImageHdr       Pointer to fw mgmt capsule Image header
+  @param[in]  FwPolicy       Firmware update policy
+  @param[out] SvnStatus      Svn compare status
+
+  @retval  EFI_SUCCESS      SVN check successful.
+  @retval  other            error occurred during firmware update
+**/
+EFI_STATUS
+CheckSblConfigDataSvn (
+  IN   EFI_FW_MGMT_CAP_IMAGE_HEADER  *ImageHdr,
+  IN   FIRMWARE_UPDATE_POLICY         FwPolicy,
+  OUT  UINT8                         *SvnStatus
+  )
+{
+  CDATA_BLOB                *CfgBlobCapAddr;
+  CDATA_BLOB                *CfgBlobFlashDataPtr;
+  UINT32                     CfgBlobFlashAddr;
+  UINT32                     CfgBlobSize;
+  EFI_STATUS                 Status;
+
+  Status = EFI_INVALID_PARAMETER;
+
+  if ((ImageHdr == NULL) || (SvnStatus == NULL)) {
+    return Status;
+  }
+
+  //
+  // Get base address of CFGDATA from  firmware block to update
+  //
+
+  if (FwPolicy.Fields.UpdatePartitionB == 0x1) {
+    Status = GetComponentInfoByPartition(FLASH_MAP_SIG_CFGDATA, TRUE, (UINT32 *) &CfgBlobFlashAddr, &CfgBlobSize);
+  } else if (FwPolicy.Fields.UpdatePartitionA == 0x1) {
+    Status = GetComponentInfoByPartition(FLASH_MAP_SIG_CFGDATA, FALSE, (UINT32 *) &CfgBlobFlashAddr, &CfgBlobSize);
+  }

```

Comment:
This logic is different with the one to update the whole bios region on reading partition A/B. should we make them aligned?

Reply:
Yes, It looks BIOS region update is getting FV version from current firmware.  It need to be checked for region capsule update would happen.